### DISCUSSION TITLE
Peer dependencies

### DISCUSSION THREAD

**Discussion by nandorojo** [expo/router#50] [Q&A] [answered]
I had some issues trying to install `expo-router` into an existing (production 😉) app. It mostly came down to clashing dependencies. Namely – if you install different versions of React Navigation, then the navigation context could differ across components, since they are different context variables. The solution is to use a yarn resolution to match the dependencies of `@react-navigation/native`.

However, I'm thinking that the `@react-navigation` dependencies should be peer dependencies of `expo-router`, rather than direct dependencies, to avoid these issues.

https://github.com/expo/router/blob/main/packages/expo-router/package.json#L46-L61

**Comment by trin4ik** [comment]
`expo/router` not for production right now. and it seems to me, in future `expo/router` replace navigations for apps (who use `expo/router`) with no backward  for `react-navigation`. i can be wrong

**Reply by nandorojo** [reply]
only way to make it ready is to use it

**Reply by trin4ik** [reply]
@nandorojo ... use it without `react-navigation` as neighbor ;)

**Reply by nandorojo** [reply]
not sure what you mean exactly, but it seems very coupled to react navigation

**Reply by trin4ik** [reply]
i mean production version `expo/router` replace `react-navigation`. and its mean no needed add `react-navigation` to the projects

**Comment by EvanBacon** [comment, marked as answer]
For now, I think we want to aim toward not having the developer install all the dependencies, this is a pretty unfortunate paradigm in React Navigation. Yarn resolutions are the way around this. Generally speaking though, changing the expected version may break Expo Router which would happen with peer dependencies or resolutions.

> I had some issues trying to install expo-router into an existing (production 😉) app

D:< Fernando!!1! lol

**Reply by nandorojo** [reply]
Makes sense. Could be worth documenting for `Existing apps` to should remove the `@react-navigation` packages installed by `expo-router`.
